internal/pkg/cosign/payload/signer.go

```diff
@@ -38,40 +38,57 @@ var _ cosign.Signer = (*payloadSigner)(nil)
 
 // Sign implements `Signer`
 func (ps *payloadSigner) Sign(ctx context.Context, payload io.Reader) (oci.Signature, crypto.PublicKey, error) {
-	payloadBytes, err := io.ReadAll(payload)
+	payloadBytes, sig, pk, err := ps.signPayload(ctx, payload)
```

Comment:
I'd preface this with `payloadBytes, err := io.ReadAll(payload)` and pass `bytes.NewBuffer(payloadBytes)` to `newSigner`

Reply:
you mean `signPayload`?
done

Reply:
yes thanks